### ISSUE TITLE
Mean image for the pre-trained net?

### ISSUE THREAD

**Issue by murbard** [uoguelph-mlrg/theano_alexnet#15]
You conveniently link to a pre-trained net, but it omits the mean image used when training it. Could you make it available?

**Comment by hma02** [comment]
Hi @murbard 
As the pre-trained net is trained on the ImageNet data, and the mean-image used when training it is generated from the data pre-processing code we provided, you can also try generating it by executing "python make_hkl.py" in the preprocessing folder. For your convenience, I put one copy of img_mean.npy in my dropbox here(https://www.dropbox.com/s/qykktrs5vyhpbov/img_mean.npy?dl=0). When you have this file in the correct path and execute python train.py or train_2gpu.py, the img_mean.npy file will be loaded. And if you also selected resume_train, the pre-trained weights will be loaded as well.

However, if you don't have the Image-net raw image and plan to use the pre-trained net to train on other dataset, you may want to generate the mean image according to your specific data set.

**Comment by murbard** [comment]
Thanks! I'll try it out. One more thing, I've rewritten the layers so that
the net is fully convolutional. For instance, if I run it on a 1024x1024
input, I get a 1000x25x25 output. Any ideas on how to make the mean image
work in that context?

I'm thinking of scaling the mean image to fit the size of the input, but I
wonder if there's a more natural extension.

**Comment by hma02** [comment]
As far as I know, AlexNet works on an input Image size of 256x256 (followed by randomly cropping down to 227x227). So the img_mean.npy provided contains average image of size 256x256. Scaling up this image to 1024x1024 would be okay but it's kind of blurry since there's not enough detailed information in 256x256.

If I understand it right, you are adjusting the input size of AlexNet from 256x256 to 1024x1024. If so, I would suggest regenerating a new average image that fits this shape and contains enough information needed for 1024x1024 sized input.

**Comment by murbard** [comment]
Thanks! Are there any other operations to be performed on the input (normalization, change of color space, etc) or is it simply (R,G,B) values in [0..255]^3 minus the offset image?

**Comment by hma02** [comment]
Hi @murbard ,
For now, beside this, we also do one random cropping of size 224x224 for each 256x256 image.
However, to further reduce error rate, we plan to add other pre-processing features in the loading process (in effect, computationally free)  mentioned in this paper: http://www.cs.toronto.edu/~fritz/absps/imagenet.pdf.
1. We plan to do ten cropping of size 224x224 for each 256x256 image to augment data and further reduce overfitting.
2. We plan to do RGB regularization on image pixels to make the Neural Network invariant  to changes in the intensity and color of the illumination.
